src/Integration.Vsix/QuickInfo/SonarLintQuickInfoControllerProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    [Export(typeof(IIntellisenseControllerProvider))]
+    [Name("SonarLint ToolTip Controller")]
+    [ContentType("text")]
+    internal class SonarLintQuickInfoControllerProvider : IIntellisenseControllerProvider
+    {
+        [Import]
+        internal IQuickInfoBroker QuickInfoBroker { get; set; }
```

Comment:
Is it required to be a property with the import attribute? Can't we use `[ImportingConstructor]` and a read only property?

Reply:
It's not required to be a property - importing in the constructor would work as well. It's more a stylistic difference in this case, and we use importing constructors everywhere else.